BlueprintUICommonControls/Sources/TransitionContainer.swift

```diff
@@ -6,14 +6,22 @@ import UIKit
 /// disappears, or changes layout.
 public struct TransitionContainer: Element {
 
-    public var appearingTransition: VisibilityTransition = .fade
-    public var disappearingTransition: VisibilityTransition = .fade
-    public var layoutTransition: LayoutTransition = .specific(AnimationAttributes())
+    public var appearingTransition: VisibilityTransition
+    public var disappearingTransition: VisibilityTransition
+    public var layoutTransition: LayoutTransition
 
     public var wrappedElement: Element
 
-    public init(wrapping element: Element) {
+    public init(
+        wrapping element: Element,
+        appearingTransition: VisibilityTransition = .fade,
```

Comment:
I considered it, though refrained as It appeared that we typically reserved `on` usage for closures that described event handling.

That said, semantically, there's a strong connection between these closures and these animations. I'll update.

Reply:
Update: it looks like our convention is to reuse the property name at the init site (checked `Box`, `Image`, `Button`, and `Rule`).

I'm going to use the `on` syntax for the extension on `Element` and leave this be for now. I'm happy to update it however if you think it's best.

Reply:
Up to you, though the convention isnt really set in stone either way; it just so happens that most parameters don't convey actions. Note it's also Swift's convention to leave type names off of parameters (eg children instead of childViewControllers), because the type system conveys that for us.